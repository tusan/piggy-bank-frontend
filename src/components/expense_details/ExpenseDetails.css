.ant-table {
  font-size: 12px;
}
.ant-modal-confirm-body .ant-modal-confirm-content {
  margin-left: 0px;
}
