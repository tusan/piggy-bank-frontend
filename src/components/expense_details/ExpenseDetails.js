import React from 'react';
import { Table } from 'antd';
import './ExpenseDetails.css';

const ExpenseDetails = props => {
  const { item } = props;

  return (
    <Table
      dataSource={[
        {
          key: 1,
          prop: 'Date',
          value: Intl.DateTimeFormat().format(item.date)
        },
        {
          key: 2,
          prop: 'Amount',
          value: 'â‚¬' + item.amount
        },
        {
          key: 3,
          prop: 'Type',
          value: item.type
        },
        {
          key: 4,
          prop: 'Description',
          value: item.description
        }
      ]}
      columns={[
        {
          title: 'prop',
          dataIndex: 'prop',
          key: 'prop',
          render: text => <b>{text}</b>,
          width: '120px'
        },
        { title: 'value', dataIndex: 'value', key: 'value' }
      ]}
      showHeader={false}
      pagination={false}
      size="small"
    />
  );
};

export default ExpenseDetails;
