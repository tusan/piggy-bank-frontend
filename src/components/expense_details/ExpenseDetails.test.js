import React from 'react';
import renderer from 'react-test-renderer';
import ExpenseDetails from './ExpenseDetails';
import moment from 'moment';

describe('<ExpenseDetails />', () => {
  it('should display an ExpenseDetails table', () => {
    const fakeExpense = {
      key: 0,
      amount: -100,
      type: 'MOTO',
      description: 'a description',
      date: moment('20181226')
    };

    const table = renderer.create(<ExpenseDetails item={fakeExpense} />);

    expect(table).toMatchSnapshot();
  });
});
