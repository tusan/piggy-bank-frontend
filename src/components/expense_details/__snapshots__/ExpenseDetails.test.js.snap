// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseDetails /> should display an ExpenseDetails table 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-small ant-table-without-column-header ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "120px",
                      "width": "120px",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <b>
                      Date
                    </b>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    2018-12-26
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={2}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <b>
                      Amount
                    </b>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    â‚¬-100
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={3}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <b>
                      Type
                    </b>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    MOTO
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={4}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <b>
                      Description
                    </b>
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    a description
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
